modin/logging/logger_base_class.py

```diff
@@ -0,0 +1,55 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""
+Module contains ``LoggerBase`` class.
+
+``LoggerBase`` is used for adding logging to Modin classes and their subclasses.
+"""
+from typing import Optional
+
+from .logger_function import logger_decorator
+
+
+class LoggerBase:
+    """
+    Ensure all subclasses of the class being inherited are logged, too.
+
+    Notes
+    -----
+    This mixin must go first in class bases declaration to have the desired effect.
+    """
+
+    def __init_subclass__(
+        cls,
+        modin_layer: str = "PANDAS-API",
+        class_name: Optional[str] = None,
+        log_level: Optional[str] = "info",
+        **kwargs,
+    ) -> None:
+        """
+        Apply logging decorator to all children of LoggerBase.
+
+        Parameters
+        ----------
+        modin_layer : str, default: "PANDAS-API"
```

Comment:
Do you want to document the `cls` parameter as well?

Reply:
That (theoretically) be done, but doc checker complains if I do so:
```
INFO:Note: NumExpr detected 12 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.
INFO:NUMPYDOC OUTPUT FOR modin\logging\logger_base_class.py
ERROR:modin.logging.logger_base_class.LoggerBase.__init_subclass__:33:PR02:Unknown parameters {'cls'}
ERROR:INVALID DOCUMENTATION FOUND
```

the reason is that `__init_subclass__` is implicitly converted to classmethod.

Reply:
agreed, added